src/js/twitter-api.js

```diff
@@ -0,0 +1,24 @@
+/*
+ * Fetches tweets from home timeline
+ *
+ * Returns a promise wrapping tweets
+ */
+export function fetchHomeTimeline() {
+	return fetch("http://localhost:8080/api/1.0/twitter/timeline").then(response => { // Attempt to fetch tweets
+		if (!response.ok) {
+			throw Error("Failed to fetch tweets from home timeline. Server responded with status code: " +
+					response.status + ", error message: " + response.statusText);
+		}
+		return response.json();
+	}).then(tweets => { // Got JSON
+		tweets.map(tweet => { // Populate default user to avoid reference error
+			if (!tweet.user) {
+				tweet.user = {
+					name: "Unknown User"
+				}
+			}
+			return tweet;
+		});
+		return tweets;
```

Comment:
if you added some default users, what is tweets here?

Reply:
Tweets is still an array of tweet objects, but the "default" users would only have a name. I added empty Strings for displayHandle and profileImageUrl to help avoid errors down the road.

Reply:
did you test to make sure this happens?

Reply:
Yes, but maybe I'm misunderstanding what you mean by "default user". I tested adding tweets with null/undefined users and the default values populated fine. Are you referring to a different case?